src/llama-hparams.h

```diff
@@ -115,6 +115,17 @@ struct llama_hparams {
     uint32_t ssm_d_state = 0;
     uint32_t ssm_dt_rank = 0;
     uint32_t ssm_n_group = 0;
+    uint32_t ssm_head_dim   = 0;
+    uint32_t ssm_mamba_d_ssm = 0;
+
+    uint32_t attn_head_dim            = 0;
```

Comment:
This parameter should can be avoided.

See the logic here:

https://github.com/ggml-org/llama.cpp/blob/a3403ae3502b5f4d562b727bca8c810a3b212198/src/llama-model.cpp#L529-L538

And as an example how we apply it for the Gemma model which can have a custom attention head size like in your case:

https://github.com/ggml-org/llama.cpp/blob/a3403ae3502b5f4d562b727bca8c810a3b212198/src/llama-model.cpp#L1021-L1025